packages/credential-provider-process/src/index.ts

```diff
@@ -9,36 +9,16 @@ import { exec } from "child_process";
  */
 export const ENV_PROFILE = "AWS_PROFILE";
 
-export interface FromProcessInit extends SharedConfigInit {
-  /**
-   * The configuration profile to use.
-   */
-  profile?: string;
-
-  /**
-   * A promise that will be resolved with loaded and parsed credentials files.
-   * Used to avoid loading shared config files multiple times.
-   *
-   * @internal
-   */
-  loadedConfig?: Promise<SharedConfigFiles>;
-}
+export interface FromProcessInit extends SourceProfileInit {}
 
 /**
  * Creates a credential provider that will read from a credential_process specified
  * in ini files.
  */
-export function fromProcess(init: FromProcessInit = {}): CredentialProvider {
-  return () =>
-    parseKnownFiles(init).then((profiles) => resolveProcessCredentials(getMasterProfileName(init), profiles, init));
-}
+export const fromProcess = (init: FromProcessInit = {}): CredentialProvider => () =>
+  parseKnownFiles(init).then((profiles) => resolveProcessCredentials(getMasterProfileName(init), profiles));
```

Comment:
nit: remove `.then()`
```suggestion
export const fromProcess = (init: FromProcessInit = {}): CredentialProvider => async () => {
  const profiles = await parseKnownFiles(init);
  return resolveProcessCredentials(getMasterProfileName(init), profiles);
};
```